### PR TITLE
Console functionalities

### PR DESCRIPTION
**Pull request by Alberaan**

### CODE REVIEW

File: consoleapp.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from alonerpg.tablesreader import *
+
+def print_data(data):
```
Comment (jrclaramunt) [inline consoleapp.py:4]: PEP8: Two blank lines between imports and functions

File: consoleapp.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    if isinstance(data, str):
+        print(data)
+        return
+    if isinstance(data, list):
+        for table in data:
+            if isinstance(table, Table):
+                print(str(table.index) + ": (" + table.system + ") " + table.filename)
+    
+    print("")
+
```
Comment (jrclaramunt) [inline consoleapp.py:14]: PEP8: Two blank lines between functions

File: consoleapp.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+def main():
+    print("Bienvenido al programa de tablas aleatorias")
+    tables = get_tables()
+    print(ayuda().data)
+    while(1):
```
Comment (jrclaramunt) [inline consoleapp.py]: Parenthesis are not needed here `while 1`

File: consoleapp.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        for table in data:
+            if isinstance(table, Table):
+                print(str(table.index) + ": (" + table.system + ") " + table.filename)
+    
+    print("")
+
+def main():
+    print("Bienvenido al programa de tablas aleatorias")
+    tables = get_tables()
+    print(ayuda().data)
```
Comment (jrclaramunt) [inline consoleapp.py:19]: Function name in Spanish

File: consoleapp.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    while(1):
+        print("Qué quieres hacer?:")
+        command = input()
+        datos = dynamic_call(tables, command)
```
Comment (jrclaramunt) [inline consoleapp.py]: Variable name should be in English. Maybe `response` ?

File: tvtropesTableCreator.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+#py
+# -*- coding: utf-8 -*-
+import requests
+from bs4 import BeautifulSoup
+import sys
+
+def createFile(title, contents):
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: PEP8: Two blank lines between imports and functions

File: tvtropesTableCreator.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    fichero = title + ".txt"
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: Variable name should be in English

File: tvtropesTableCreator.py (hunk 6 above)
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: PEP8: Function names should be written following the conventon `lower_case_with_underscores`

File: tvtropesTableCreator.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    print("Escribiendo en " + fichero)
+    with open(fichero, "w", encoding="utf-8") as fileToWrite:
+        fileToWrite.write(contents)
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: PEP8: Function names should be written following the conventon lower_case_with_underscores

File: tvtropesTableCreator.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    print("Terminado")
+
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py:14]: PEP8: Two blank lines between functions

File: tvtropesTableCreator.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    titleToReturn = ""
+    for i in title:
+        if i.isalnum() or i == " ":
+            titleToReturn += i
+        else:
+            titleToReturn += " "
+
+    return titleToReturn
+
+def getTropes(url):
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: PEP8: Two blank lines between functions

File: tvtropesTableCreator.py [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            toAdd = trope.get_text().split("/")[0]
+
+            if any(char.isalnum() for char in toAdd): 
+                text += toAdd + "\n"
+
+    return title, text[:-1] # I remove the last carriage return of the text
+
+def main():
+    if (sys.argv[1] == "-i"):
+        while(1):
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: Unneded parenthesis here as well

File: consoleapp.py [hunk 12]
```diff
@@ -0,0 +1,27 @@
+from alonerpg.tablesreader import *
+
+def print_data(data):
+    if isinstance(data, str):
```
Comment (jrclaramunt) [inline consoleapp.py:5]: This could actually be a good opportunity to get rid of the ifs, but maybe we can leave it out od the scope of this PR

**Review by jrclaramunt — CHANGES_REQUESTED** [review]
Good job! Some PEP8 fixing mostly

File: consoleapp.py (hunk 2 above)
Comment (Alberaan) [reply consoleapp.py:14]: This is not a function

File: consoleapp.py (hunk 4 above)
Comment (Alberaan) [reply consoleapp.py:19]: Not changing it. This is not only a function name, but also a command that the final user will type in the terminal, so it needs to be in spanish

File: consoleapp.py (hunk 12 above)
Comment (Alberaan) [reply consoleapp.py:5]: Next PR then. I'm not sure how to remove the else

File: tvtropesTableCreator.py [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+def create_file(title, contents):
+    myfile = title + ".txt"
+    print("Escribiendo en " + myfile)
+    with open(myfile, "w", encoding="utf-8") as file_to_write:
+        file_to_write.write(contents)
+    print("Terminado")
+
+
+def fix_title(title):
+    titleToReturn = ""
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py:17]: naming convention

File: tvtropesTableCreator.py [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+    return titleToReturn
+
+
+def get_tropes(url):
+    tvtropes_link = url
+    url ='http://dbtropes.org/info?uri=' + tvtropes_link + "&submit=Reverse+URI+Lookup"
+    page = requests.get(url)
+    soup = BeautifulSoup(page.text, "html.parser")
+    unfilteredTitle = soup.select("head title")[0].get_text().replace(" - DBTropes","")
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: naming convention

File: tvtropesTableCreator.py [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+    title = fix_title(unfilteredTitle)
+    
+    tropes = soup.select("tr.bg1 a")
+
+    text = ""
+
+    for trope in tropes:
+        if "Main" in trope["href"]:
+            toAdd = trope.get_text().split("/")[0]
```
Comment (jrclaramunt) [inline tvtropesTableCreator.py]: naming convention

**Review by jrclaramunt — CHANGES_REQUESTED** [review]

**Review by jrclaramunt — APPROVED** [review]
Some refactor to be done, but let's keep it out of the scope of this PR

**Pull request merged**